test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusMainTestExtension.java

```diff
@@ -130,6 +129,35 @@ public void afterEach(ExtensionContext context) throws Exception {
         result = null;
     }
 
+    private static void setFinalStatic(Class clazz, Field field, Object newValue) throws Exception {
+        // WORKAROUND to prevent:
+        // "An illegal reflective access operation has occurred" warning to be shown
+        // this hack adds: --add-opens
+        // from: https://stackoverflow.com/questions/46454995/how-to-hide-warning-illegal-reflective-access-in-java-9-without-jvm-argument#comment110253314_46458447
+        clazz.getModule().addOpens(clazz.getPackageName(), QuarkusMainTestExtension.class.getModule());
```

Comment:
This seems to be causing test failures.

Reply:
I discovered that this approach unfortunately doesn't work with Java >= 12.

Attempted with a different approach here: https://github.com/quarkusio/quarkus/pull/22455/commits/3168b210ffc9407ff32caa32498a2fd79f7ce318

please have a second review @stuartwdouglas since here we are "reversing" the logging handlers injected by Quarkus itself.

Reply:
cc. @geoand since your approval is based on a different implementation.